exercises/tutorial_halfday/ex3_colored-indexset_solution.cpp

```diff
@@ -264,14 +272,32 @@ int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
 
 #if defined(RAJA_ENABLE_CUDA)
 
+//
+// Resource object used to construct list segment objects with indices
+// living in host (CPU) memory.
+//
+  camp::resources::Resource cuda_res{camp::resources::Cuda()};
+
+//
+// Create a RAJA TypedIndexSet with four ListSegments, one for the indices of
+// the elements in each subsut. This will be used in the RAJA OpenMP and CUDA
+// variants of the vertex sum calculation.
+
+  RAJA::TypedIndexSet<SegmentType> cuda_colorset;
+
+  cuda_colorset.push_back( SegmentType(&idx[0][0], idx[0].size(), cuda_res) );
+  cuda_colorset.push_back( SegmentType(&idx[1][0], idx[1].size(), cuda_res) );
+  cuda_colorset.push_back( SegmentType(&idx[2][0], idx[2].size(), cuda_res) );
+  cuda_colorset.push_back( SegmentType(&idx[3][0], idx[3].size(), cuda_res) );
+
   std::cout << "\n Running RAJA CUDA index set vertex sum...\n";
 
   std::memset(areav, 0, Nvert*Nvert * sizeof(double));
 
   using EXEC_POL4 = RAJA::ExecPolicy<RAJA::seq_segit, 
                                      RAJA::cuda_exec<CUDA_BLOCK_SIZE>>;
 
-  RAJA::forall<EXEC_POL4>(colorset, [=] RAJA_DEVICE (int ie) {
+  RAJA::forall<EXEC_POL4>(cuda_colorset, [=] RAJA_DEVICE (int ie) {
```

Comment:
Is  ``cuda_colorset`` used to specify a stream?

Reply:
No. Since we now pass a resource to the list segment ctor, we can't use the same list segment in both host and device kernels. We need a different list segment for each.